.github/scripts/validate_new_chain_request.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+
+"""
+This script is used to validate a GitHub issue for a new chain request.
+What it does:
+1. Extracts the RPC URL from the issue body
+2. Extracts the chain ID from the RPC URL
+3. Checks if the chain ID is in chainlist
+4. Checks if the factory is already deployed
+5. Checks if the deployer address is pre-funded
+
+If any of the above checks fails, the script will exit with an error code and set an environment variable
+`COMMENT_OUTPUT` with an error message. The GitHub action will then use this variable to post a comment.
+For the above reason, the script needs to be executed in the GitHub's action shell, for example:
+```
+. ./.github/scripts/validate_new_chain_request.sh
+```
+
+The script expects the following environment variables to be set:
+- `ISSUE_BODY` - the body of the issue
+- `FACTORY_ADDRESS` - the address of the factory contract
+- `FACTORY_DEPLOYER_ADDRESS` - the address of the factory deployer
+- `FACTORY_BYTECODE` - the bytecode of the factory contract
+
+"""
+
+ERROR_MSG_RPC_FAILURE="""
+**⛔️ Error:**<br>
+Could not fetch the chain ID from the RPC URL.<br>
+Either the RPC URL is missing, or the RPC is not responding.<br>
+**Make sure your issue follows the issue template and includes the RPC URL.**<br>
+:sparkles: You can edit the issue to trigger the check again. :sparkles:"""
+ERROR_MSG_CHAINLIST_FAILURE="""
+**⛔️ Error:**<br>
+Couldn't find the chain in the chainlist. Please make sure it's added to the chainlist.<br>
+For more information on how to add a chain, please refer to the [chainlist repository](https://github.com/ethereum-lists/chains).<br>
+:sparkles: You can edit the issue to trigger the check again. :sparkles:"""
+ERROR_MSG_PREFUND_CHECK="""
+**⛔️ Error:**<br>
+There was an error while estimating the deployment transaction and checking the pre-fund. Please make sure that the RPC URL is valid and reachable.<br>
+:sparkles: You can edit the issue to trigger the check again. :sparkles:"""
+SUCCESS_MSG="""
+**✅ Success:**<br>
+The issue description is valid:<br>
+- The RPC URL is valid<br>
+- The chain is in the chainlist<br>
+- The deployer address is pre-funded<br>
+:sparkles: The team will be in touch with you soon :sparkles:"""
+
+
+trim() {
+    # 3.5.3 Shell Parameter Expansion at https://www.gnu.org/software/bash/manual/bash.html
+    local var="$*"
+    # remove leading whitespace characters
+    var="${var#"${var%%[![:space:]]*}"}"
+    # remove trailing whitespace characters
+    var="${var%"${var##*[![:space:]]}"}"
+    echo "$var"
+}
+
+rpc_url=$(echo $ISSUE_BODY | egrep -o 'https?://[^ ]+' -m 1 | head -1)
+
+if [ -z "$rpc_url" ]; then
+    echo "COMMENT_OUTPUT=\"$ERROR_MSG_RPC_FAILURE\"" >> $GITHUB_ENV
+    exit 1
+fi
+
+rpc_url="$(trim $rpc_url)"
+echo "Extracted RPC url: $rpc_url. Trying to get the chain id..."
+response=$(curl $rpc_url --location --header \
```

Comment:
```suggestion
response=$(curl "$rpc_url" --location --header \
```

Reply:
Actually never mind - quoting only matters if it can contain spaces or be empty.